### ISSUE TITLE
Cannot find module ref in dev mode

### ISSUE THREAD

**Issue by mjangir** [chentsulin/electron-react-boilerplate#1757]
As per your suggestion, I installed some of my native dependencies like **node ffi** and **ref**  through the app package.json. I just required them where ever I wanted in my react application. The problem is they just work  fine in the production build but not in the development mode.

In dev mode, it is saying **Error: Cannot find module 'ref'**. Though the webpack is marking it as external dependency and converts the syntax to **var ref = require('ref')** perfectly fine but still the application is not able to load it from **app/node_modules**. 

Even when I tried to debug it, the **paths** array in **module.js** doesn't contain the path of **app/modules** where it should search the dependency. 

Could anyone help me out in this? Really appreciated.

**Comment by amilajack** [comment]
Interesting. Can you please share the entire stack trace. Take a picture of your devtool logs.

**Comment by mjangir** [comment]
Actually I was using html-webpack-plugin to create the HTML file dynamically and in Dev mode obiviously it is served from memory. In the Dev mode, I was loading the application with http protocol in main.dev.js and then I got to know that nodejs module resolution only works on file protocol.

Finally I used html-webpack-harddisk-plugin which writes the HTML file to disk and I loaded that with file protocol again and it worked.

Thanks
